### PR TITLE
fixes #156 to detect version of consent

### LINKED ISSUE #156: It does not detect version of consent
**Version**
1.0.1-3

**Module (core, cmpapi, cli, stub, or testing)**
core

**Describe with reproduction steps – What is the expected behavior?**
I tried to decode consent of first version.
```js
TCString.decode('BOhwdphOxFC7tAHABBFRC--AAAAuhr_7__7-_9_-_f__9uj3Or_v_f__32ccL59v_h_7v-_7fi_20nV4u_1vft9yfk1-5ctDztp507iakivXmqdeb9v_nz3_5pxP78k89r7337Ew_v8_v-b7BCON_YxEiA');
```

I got an error:

> TCModelError: invalid value ` passed for publisherCountryCode

### PR DESCRIPTION
**Pull request by marliotto**
#156

### CODE REVIEW

File: modules/core/src/encoder/SegmentEncoder.ts [hunk 1]
```diff
@@ -77,16 +77,23 @@ export class SegmentEncoder {
...
     const bitField = Base64Url.decode(encodedString);
     let bStringIdx = 0;
 
+    if (segment === Segment.CORE) {
+
+      tcModel.version = IntEncoder.decode(bitField.substr(bStringIdx, BitLength[Fields.version]), BitLength[Fields.version]);
+
+    }
+
     if (segment !== Segment.CORE) {
```
Comment (chrispaterson) [inline modules/core/src/encoder/SegmentEncoder.ts:89]: Can we just make this an `else` to the `if` that you're adding since this is the other case?

Comment (marliotto) [reply modules/core/src/encoder/SegmentEncoder.ts:89]: I thought about it.
However, there are two different case.
First case is detecting version.
Second case refers to segment type detecting.
Two "if" are better for future refactoring.

Comment (chrispaterson) [reply modules/core/src/encoder/SegmentEncoder.ts:89]: Fair enough...

**Comment by chrispaterson** [comment]
Thanks again!

**Comment by chrispaterson** [comment]
in version 1.0.1-4

**Pull request merged**